### DISCUSSION TITLE
PSF Grant for reviewing and releasing Python 3 support for Pillow

### DISCUSSION THREAD

**Discussion by aclark4life** [python-pillow/Pillow#8221] [General]
In order to find some time to finish reviewing and releasing Pillow for Python 3, I'm considering asking the PSF for some grant money to fund my development time. But I want to "do it right" so I'm asking for feedback here. E.g. @fluggo if the PSF approves my request can I forward some of the $$$ to you since you did the majority of the coding? If so, can you estimate how much time you put in, and also specify a conservative hourly rate estimate? E.g. "$50/hour". Would anyone else like to be compensated based on their contributions and pending a grant award from the PSF? Ready… go!

**Comment by fluggo** [comment]
Thanks very much for the offer, but I'm not interested in compensation. I'm
just happy my contribution was useful.

--Brian

**Comment by aclark4life** [comment]
Cool thanks for the reply! Just wanted to be sure :+1:

**Comment by aclark4life** [comment]
Anyone else YAY or NAY on this?

**Comment by warsaw** [comment]
As an interested bystander, I fully support you getting grant money to help get an official Python 3 compatible Pillow release out the door!

**Comment by aclark4life** [comment]
Cool, thanks @warsaw !

**Comment by pydanny** [comment]
Piillow and PIL are a critical part of python 3 adoption. Its a blocked as much ad or more so than Django. Ask for the money.

**Comment by aclark4life** [comment]
Thanks @pydanny!

**Comment by gora** [comment]
We would definitely be interested in seeing Pillow developed. So, +1 from our side for this to be funded.

**Comment by strichter** [comment]
The lack of PIL (Pillow) on Python 3 is a major reasons several larger scale projects cannot be ported yet.

Very simple example: We use Dozer with some additions to analyze/profile our WSGI app, but without Pillow on Python 3, Dozer does not work.

**Comment by aclark4life** [comment]
Thanks for the comments @gora and @strichter

**Comment by d-schmidt** [comment]
I would take an hour or two as compensation, as a student I'm always in need of money but I don't really care. I didn't fix it in my free time. (If you need a proper number to tell them, I'm at 20-30 hours slowly figuring out how various thing works in PIL and how to fix them)
Pull all the fixes waiting and ask them. It is an important project and worth it and I'm glad somebody pulled the string to bring PIL back to life.

**Comment by aclark4life** [comment]
@d-schmidt Thanks for the comment and your contributions! OK so if I understand you correctly you have spent at least 20-30 hours and are asking for 1-2 hours of compensation? Are you in the USA or elsewhere?

**Comment by d-schmidt** [comment]
Germany. And yes, 30h \* 50$? That would be a crazy amount of money. I wouldn't deny it, but a week of food would be awesome enough. It is voluntary all along and I've need the patches myself. :)

**Comment by aclark4life** [comment]
Let me see what I can do. I don't feel comfortable asking for 30h on your behalf, but I can ask for 1-2 hours @ $50 USD. I'm asking for something like 24 hours @ $50, which is only a small fraction of the total time I have invested and includes only the estimated time I need to finish preparing the next release. You can always ask the PSF for additional funding, though: http://www.python.org/psf/grants/. Assuming you are OK with all of this, we could theoretically PayPal you the $$$ once we receive a check from the PSF (in the event our funding request is approved, of course).

**Comment by d-schmidt** [comment]
@aclark4life sounds great.

**Comment by aclark4life** [comment]
Proposal I will send tomorrow, comments welcome: https://github.com/python-imaging/psf-grant-proposal

**Comment by aclark4life** [comment]
@d-schmidt Resolved by the PSF board in our favor! I just invoiced the PSF, so as soon as we receive the check I'll PayPal you $100 USD. Thanks for helping with Pillow.

**Comment by d-schmidt** [comment]
Thanks. I will happily continue to do so :)

**Comment by gora** [comment]
Wonderful news! Congratulations on getting PSF support for Pillow.

**Comment by akaIDIOT** [comment]
:+1: on getting compensated for your great work, guys. Having an imaging lib for py3 must really help the adoption. I know it does for me :)

**Comment by aclark4life** [comment]
Thanks @gora and @akaIDIOT!

**Comment by aclark4life** [comment]
Hey @d-schmidt can you email me (aclark@aclark.net) with your preferred PayPal email address? Will send payment this week sometime.

**Comment by aiwonderland** [comment]
Hello from 2026!

**Reply by hugovk** [reply]
@aiwonderland Please don't spam old discussions.
